loyalty_multi_gift/models/loyalty_reward.py

```diff
@@ -0,0 +1,69 @@
+# Copyright 2021 Tecnativa - David Vidal
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo import _, api, fields, models
+
+
+class LoyaltyProgram(models.Model):
+    _inherit = "loyalty.program"
+
+    loyalty_multi_gift_ids = fields.One2many(
+        comodel_name="loyalty.reward.product_line",
+        inverse_name="program_id",
+        string="Gift list",
+    )
+    program_type = fields.Selection(
+        selection_add=[("multi_gift", "Multi Gift")],
+        ondelete={"multi_gift": "set default"},
+    )
+
+    def name_get(self):
+        """Add complete description for the multi gift reward type."""
+        res = super().name_get()
+        for reward in self.filtered(lambda x: x.program_type == "multi_gift"):
+            reward_string = _("Free Products - %(name)s") % {
+                "name": ", ".join(
+                    f"{reward.reward_product_quantity}x "
+                    f"{fields.first(reward.reward_product_ids).name}"
+                    for reward in reward.loyalty_multi_gift_ids
+                )
+            }
+            res.append((reward.id, reward_string))
+        return res
+
+
+class LoyaltyGift(models.Model):
+    _name = "loyalty.reward.product_line"
+    _description = "Loyalty Multi Gift"
+
+    program_id = fields.Many2one(comodel_name="loyalty.program")
+    reward_product_quantity = fields.Integer(
+        string="Quantity",
+        help="Reward product quantity",
+    )
+    reward_default_product_id = fields.Many2one(
+        comodel_name="product.product",
+        compute="_compute_reward_default_product_id",
+        inverse="_inverse_reward_default_product_id",
+        readonly=False,
+    )
+    reward_product_ids = fields.Many2many(
```

Comment:
Odoo already can make gifts optional, although the scheme is different from ours. Here, for every line we could give the customer the option to choose which product to reward while Odoo offers the option of which reward (as a whole) to choose. It would be very interesting to converge although I'm not sure if we can make it happen (maybe we find limitations as we did in the multi criteria module...)